Added a dictionary extension method ConvertValueToNewType (#21)
SCADAMINDS/swiss-army-knife: Documentation/Extensions.md

// Will return "myFallbackValue"
myDictionary.GetOr("nonExistingKey", () => "myFallbackValue");

> var myDictionary = new Dictionary<string, string> {

Reviewer: Let's call it `myIntDictionary` or similar so we don't confuse it with the dictionary above.

    {"myKey", "2"}
}
 // Will return a new dictionary where values type is integer

>  var convertedValueDictionary = myDictionary.ConvertValuesToNewType<string, string, int>("myKey", oldValue => int.Parse(oldValue, CultureInfo.InvariantCulture));

Reviewer: If it makes it cleaner you don't actually need to specify the type parameters.
```
var newDictionary = myDictionary.ConvertValuesToNewType(v => int.Parse(v));
```
is totally valid.

Author: yes I saw that visual studio suggested it as a change as well, I am then curious how will the method infer the type of the new dictionary? can it do automatically?

Reviewer: Yeah it can infer it from the dictionary you provide into it and the return type of the `selector` function

> // Will throw exception
> var myDictionary = new Dictionary<string, string> {
>     {"myKey", "bar"}
> }
> var throwsOnConvertDictionary = myDictionary.SelectValues("myKey", oldValue => int.Parse(oldValue, CultureInfo.InvariantCulture));

Reviewer: Suggest we just remove the new "myDictionary" comment like below.

```suggestion
// Using <string,string> dictionary from earlier with unparsable string. Will throw exception
var throwsOnConvertDictionary = myDictionary.SelectValues("myKey", oldValue => int.Parse(oldValue, CultureInfo.InvariantCulture));
```
